### PR TITLE
[Enhance](resource group)db support replication_allocation

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

- db support replication_allocation,when create table,if not set `replication_num` or `replication_allocation `,will use it in db
- fix partition property will disappear when table partition is not null

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-DATABASE.md [hunk 1]
```diff
@@ -59,6 +59,13 @@ CREATE DATABASE [IF NOT EXISTS] db_name
   - `iceberg.hive.metastore.uris` : hive metastore service address;
   - `iceberg.catalog.type`: The default is `HIVE_CATALOG`; currently only `HIVE_CATALOG` is supported, and more Iceberg catalog types will be supported in the future.
 
+- If you want to specify the default replica distribution for tables in db, you need to specify `replication_allocation` (the `replication_allocation` attribute of table will have higher priority than db)
+  ```sql
```
Comment (morningman) [inline docs/en/docs/sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-DATABASE.md:63]: add empty line before

File: docs/zh-CN/docs/sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-DATABASE.md [hunk 2]
```diff
@@ -59,6 +59,13 @@ CREATE DATABASE [IF NOT EXISTS] db_name
   - `iceberg.hive.metastore.uris` ：hive metastore 服务地址;
   - `iceberg.catalog.type`： 默认为 `HIVE_CATALOG`；当前仅支持 `HIVE_CATALOG`，后续会支持更多 Iceberg catalog 类型。
 
+- 如果要为db下的table指定默认的副本分布策略，需要指定`replication_allocation`（table的`replication_allocation`属性优先级会高于db）
+  ```sql
```
Comment (morningman) [inline docs/zh-CN/docs/sql-manual/sql-reference/Data-Definition-Statements/Create/CREATE-DATABASE.md:63]:
```sql 前添加一个空行

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java [hunk 3]
```diff
@@ -858,33 +859,35 @@ public Map<Long, TableIf> getIdToTable() {
 
     public void replayUpdateDbProperties(Map<String, String> properties) {
         dbProperties.updateProperties(properties);
-        binlogConfig = dbProperties.getBinlogConfig();
+        if (PropertyAnalyzer.hasBinlogConfig(properties)) {
+            binlogConfig = dbProperties.getBinlogConfig();
+        }
     }
 
     public boolean updateDbProperties(Map<String, String> properties) throws DdlException {
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java:867]: when init and update db property replica alloc,  better to  check the system has enough backends ?

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**